services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -123,21 +124,56 @@ export class WorkflowService {
 
   async updateWorkflowRuntimeData(workflowRuntimeId: string, data: WorkflowDefinitionUpdateInput) {
     const runtimeData = await this.workflowRuntimeDataRepository.findById(workflowRuntimeId);
-    const ajv = new Ajv();
-    // const validate = ajv.compile(runtimeData.contextSchema);
+    const workflow = await this.workflowDefinitionRepository.findById(
+      runtimeData.workflowDefinitionId,
+    );
+    const validateContextSchema = ajv.compile((workflow?.contextSchema as any)?.schema as Schema);
```

Comment:
You should only validate the context if a schema exists on the workflow definition